connect/runtime/src/main/java/org/apache/kafka/connect/runtime/RestartRequest.java

```diff
@@ -25,14 +25,12 @@
  * A request to restart a connector and/or task instances.
  * <p>
  * The natural order is based first upon the connector name and then requested restart behaviors.
- * If two requests have the same connector name, then the requests are ordered based on the 
+ * If two requests have the same connector name, then the requests are ordered based on the
  * probable number of tasks/connector this request is going to restart.
  */
-public class RestartRequest implements Comparable<RestartRequest> {
-
-    private final String connectorName;
-    private final boolean onlyFailed;
-    private final boolean includeTasks;
+public record RestartRequest(String connectorName,
```

Comment:
I suggest moving the original `@param` documentation to the class level, which would allow the removal of the getters and their associated Javadoc. The null check can be done using a compact constructor. WDYT?

For example:
```java
    public RestartRequest {
        Objects.requireNonNull(connectorName, "Connector name may not be null");
    }
```

Reply:
Good idea, that makes the code even shorter. Done!